app/services/update_component_structure.rb

```diff
@@ -0,0 +1,20 @@
+class UpdateComponentStructure
+
+  attr_reader :object_id
+
+  def self.call(*args)
+    event = ActiveSupport::Notifications::Event.new(*args)
+    payload = event.payload
+    return false if payload[:skip_structure_updates]
+    return false unless Component::STRUCTURALLY_RELEVANT_DATASTREAMS.include?(payload[:file_id])
+    return false unless model(payload[:pid]) == 'Component'
+    SetDefaultStructure.new(payload[:pid]).enqueue_default_structure_job
+  end
+
+  def self.model(repo_id)
+    SolrDocument.find(repo_id)[Ddr::Index::Fields::ACTIVE_FEDORA_MODEL]
+  rescue Ddr::Models::SolrDocument::NotFound
```

Comment:
I chose `Ddr::Models::SolrDocument::NotFound` because I see this behavior in a `rails console` in the `dul-hydra` project ...

```
2.3.1 :001 > SolrDocument.find('foo')
Ddr::Models::SolrDocument::NotFound: SolrDocument not found for "foo".
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/bundler/gems/ddr-models-dd57d201c6c8/lib/ddr/models/solr_document.rb:22:in `find'
	from (irb):1
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/gems/railties-4.2.7/lib/rails/commands/console.rb:110:in `start'
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/gems/railties-4.2.7/lib/rails/commands/console.rb:9:in `start'
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/gems/railties-4.2.7/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/gems/railties-4.2.7/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/coblej/.rvm/gems/ruby-2.3.1@dul-hydra/gems/railties-4.2.7/lib/rails/commands.rb:17:in `<top (required)>'
	from script/rails:6:in `require'
	from script/rails:6:in `<main>'
```

I will add a test for `#model` to confirm the behavior.